lib/dry/view/scope_builder.rb

```diff
@@ -0,0 +1,74 @@
+require 'dry/inflector'
+require_relative 'scope'
+
+module Dry
+  module View
+    class ScopeBuilder
+      attr_reader :namespace
+      attr_reader :rendering
+
+      def initialize(namespace: nil, rendering: nil)
+        @namespace = namespace
+        @rendering = rendering
+      end
+
+      def for_rendering(rendering)
+        return self if rendering == self.rendering
+
+        self.class.new(namespace: namespace, rendering: rendering)
+      end
+
+      def rendering?
+        !!rendering
+      end
+
+      def call(name = nil, locals)
+        scope_class(name).new(
+          name: name,
+          locals: locals,
+          rendering: rendering,
+        )
+      end
+
+      private
+
+      DEFAULT_SCOPE_CLASS = Scope
+
+      def scope_class(name = nil)
+        if name.nil?
+          DEFAULT_SCOPE_CLASS
+        elsif name.is_a?(Class)
+          name
+        else
+          resolve_scope_class(name: name)
+        end
+      end
+
+      def resolve_scope_class(name:)
+        return fallback_class unless namespace
```

Comment:
@timriley where is `fallback_class` defined?

Reply:
Ah thanks @GustavoCaso, that line shouldn't have been there!